scipy/stats/tests/test_distributions.py

```diff
@@ -4629,6 +4632,160 @@ def test_burr_nan_mean_var_9544(self):
+            18.28, 34.12, 39.69, 43.05,
+            6.487, 9.352, 10.39, 11.03,
+            5.444, 7.313, 7.966, 8.370,
+            4.772, 6.039, 6.448, 6.695,
+            4.654, 5.823, 6.191, 6.411]
+    qs = np.concatenate((q05, q01, q001))
+    ps = [.95, .99, .999]
+    data = zip(product(ps, vs, ks), qs)
+
+    path_prefix = os.path.dirname(__file__)
+    relative_path = "data/studentized_range_mpmath_ref.json"
+    with open(os.path.join(path_prefix, relative_path), "r") as file:
+        pregenerated_data = json.load(file)
+
+    @pytest.mark.parametrize("case_result", pregenerated_data["cdf_data"])
+    def test_cdf_against_mp(self, case_result):
+        src_case = case_result["src_case"]
+        mp_result = case_result["mp_result"]
+
+        res = stats.studentized_range.cdf(src_case["q"],
+                                          src_case["k"],
+                                          src_case["v"])
+
+        assert_allclose(res, mp_result,
+                        atol=src_case["expected_atol"],
+                        rtol=src_case["expected_rtol"])
+
+    @pytest.mark.parametrize("case_result", pregenerated_data["pdf_data"])
+    def test_pdf_against_mp(self, case_result):
+        src_case = case_result["src_case"]
+        mp_result = case_result["mp_result"]
+
+        res = stats.studentized_range.pdf(src_case["q"],
+                                          src_case["k"],
+                                          src_case["v"])
+
+        assert_allclose(res, mp_result,
+                        atol=src_case["expected_atol"],
+                        rtol=src_case["expected_rtol"])
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize("case_result", pregenerated_data["moment_data"])
+    def test_moment_against_mp(self, case_result):
+        src_case = case_result["src_case"]
+        mp_result = case_result["mp_result"]
+
+        res = stats.studentized_range.moment(src_case["m"],
+                                             src_case["k"],
+                                             src_case["v"])
+
+        assert_allclose(res, mp_result,
+                        atol=src_case["expected_atol"],
+                        rtol=src_case["expected_rtol"])
+
+    def test_cdf_against_tables(self):
+        for pvk, q in self.data:
+            p_expected, v, k = pvk
+            res_p = stats.studentized_range.cdf(q, k, v)
+            assert_allclose(res_p, p_expected, rtol=1e-4)
+
+    @pytest.mark.slow
+    def test_ppf_against_tables(self):
+        for pvk, q_expected in self.data:
+            p, v, k = pvk
+            res_q = stats.studentized_range.ppf(p, k, v)
+            assert_allclose(res_q, q_expected, rtol=1e-4)
+
+    def test_pdf_integration(self):
+        k, v = 3, 10
+
+        # Test whether PDF integration is 1 like it should be.
+        res = quad(stats.studentized_range.pdf, 0, np.inf, args=(k, v))
+        assert_allclose(res[0], 1)
+
+    @pytest.mark.slow
```

Comment:
Please find all the lists of slow tests in the azure logs that look like:
```
235.83s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_cont_basic[500-200-studentized_range-arg92]
83.71s call     build/testenv/lib/python3.7/site-packages/scipy/sparse/linalg/dsolve/tests/test_linsolve.py::TestSpsolveTriangular::test_random
75.72s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_moments[ksone-arg57-True-True-False]
64.65s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_moments[studentized_range-arg93-True-True-False]
42.50s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_fit.py::test_cont_fit[MLE-genhyperbolic-arg32]
41.07s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_distributions.py::TestStudentizedRange::test_pdf_against_cdf
36.99s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_stats.py::TestMGCStat::test_twosamp
36.58s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_cont_basic[500-200-ksone-arg57]
35.88s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_moments[vonmises-arg102-False-False-False]
34.23s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_distributions.py::TestLevyStable::test_pdf_nolan_samples
```
They depend a bit on platform, so please check each. I don't think we want to be on these lists. I'd suggest marking `@pytest.mark.xslow` unless they are really important, in which case let's discuss them.

Reply:
I collected all the "slowest duration" tests from azure that had "studentized" in the name: (repeats removed)
```
['stats/tests/test_continuous_basic.py::test_cont_basic[500-200-studentized_range-arg92]'
 'stats/tests/test_continuous_basic.py::test_moments[studentized_range-arg93-True-True-False]'
 'stats/tests/test_distributions.py::TestStudentizedRange::test_fitstart_valid'
 'stats/tests/test_distributions.py::TestStudentizedRange::test_pdf_against_cdf']
```

I just marked test_fitstart_valid and test_pdf_against_cdf as `@pytest.mark.xslow`. It will be in the next commit. I haven't looked closely at `test_continuous_basic.py` yet.

Reply:
We're on the list for `test_cont_basic` because of this line: 
https://github.com/scipy/scipy/blob/dd98fe5dfe5a9fe26b1cd8479cd46c67f7eed296/scipy/stats/tests/test_continuous_basic.py#L138

Where it tries to generate 500 random variates:
```
In [7]: %timeit stats.studentized_range.rvs(3, 10, size=5)
904 ms ± 70 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
.904s * 100 = 90 seconds for 500 random variates.